src/ir/handle_connection_spec.h

```diff
@@ -29,24 +29,24 @@ namespace raksha::ir {
 // equivalent ArcsManifestProtos.
 class HandleConnectionSpec {
  public:
-  explicit HandleConnectionSpec(
-    std::string name, bool reads, bool writes,
-    std::unique_ptr<types::Type> type)
-    : name_(std::move(name)), reads_(reads), writes_(writes),
-      type_(std::move(type)) {}
+  explicit HandleConnectionSpec(std::string name, bool reads, bool writes,
+                                types::Type type)
+      : name_(std::move(name)),
+        reads_(reads),
+        writes_(writes),
+        type_(std::move(type)) {}
 
   const std::string &name() const { return name_; }
   bool reads() const { return reads_; }
   bool writes() const { return writes_; }
-  const types::Type &type() const { return *type_; }
+  const types::Type &type() const { return type_; }
```

Comment:
If `Type` is a wrapper for `std::unique_ptr<TypeBase>`, this causes us to return a pointer to a pointer. I'm not worried about the performance per-se, I think though that this is perhaps a code smell of sorts.

Reply:
This is just a temporary state and will be fixed by having `Type` cheaply copyable.